FilePersistence/src/version_control/GitRepository.cpp

```diff
@@ -318,30 +318,48 @@ CommitGraph GitRepository::commitGraph(QString startRevision, QString endRevisio
 
 const Commit* GitRepository::getCommit(QString revision) const
 {
-	Q_ASSERT(revision != WORKDIR);
 	Q_ASSERT(revision != INDEX);
+	if (revision == WORKDIR)
+	{
+		Commit* m = nullptr;
+		QDirIterator DirIter(workdirPath(), QDirIterator::Subdirectories);
+		while (DirIter.hasNext())
+		{
+			DirIter.next();
+			if (DirIter.fileInfo().isFile()){
+				QFile file(DirIter.filePath());
+				QByteArray arr = file.readAll();
+				char* content = arr.data();
```

Comment:
Hmm, the memory operations here are not quite right. You are just referencing arr.data() which will be destroyed once arr gets out of scope. you need to copy it instead. See here for how to copy:
http://doc.qt.io/qt-5/qbytearray.html#data